brayns/engine/Scene.cpp

```diff
@@ -63,37 +63,44 @@ std::shared_ptr<T> _remove(std::vector<std::shared_ptr<T>>& list,
 
 namespace brayns
 {
-Scene::Scene(ParametersManager& parametersManager)
-    : _parametersManager(parametersManager)
+Scene::Scene(AnimationParameters& animationParameters,
+             GeometryParameters& geometryParameters,
+             VolumeParameters& volumeParameters)
+    : _animationParameters(animationParameters)
+    , _geometryParameters(geometryParameters)
+    , _volumeParameters(volumeParameters)
 {
 }
 
-Scene& Scene::operator=(const Scene& rhs)
+void Scene::cloneFrom(const Scene& rhs)
 {
     if (this == &rhs)
-        return *this;
+        return;
 
     {
         std::unique_lock<std::shared_timed_mutex> lock(_modelMutex);
         std::shared_lock<std::shared_timed_mutex> rhsLock(rhs._modelMutex);
-        _modelDescriptors = rhs._modelDescriptors;
+
+        _modelDescriptors.clear();
+        _modelDescriptors.reserve(rhs._modelDescriptors.size());
+        for (const auto& modelDesc : rhs._modelDescriptors)
+        {
+            auto newModelDesc =
+                std::make_shared<ModelDescriptor>(createModel(),
+                                                  modelDesc->getPath());
+            newModelDesc->cloneFrom(*modelDesc);
+            _modelDescriptors.push_back(newModelDesc);
```

Comment:
Can't you simplify lines 88:92 to `_modelDescriptors.push_back(modelDesc->clone())`?

Reply:
Probably I can.